api/Services/MachineLearningService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using api.Datasets.Models;
+using api.Model;
+using Microsoft.ML;
+
+namespace api.Services;
+
+public class MachineLearningService : IMachineLearningService
+{
+    private readonly ApplicationContext _context;
+
+    private static string? AppPath => Path.GetDirectoryName(Environment.GetCommandLineArgs()[0]);
+    private static string TrainDataPath => Path.Combine(AppPath!, "..", "..", "..", "Datasets", "training_angular-commits.tsv");
+    private static string ModelPath => Path.Combine(AppPath!, "..", "..", "..", "TrainedModels", "angular-model.zip");
+
+    private static MLContext? _mlContext;
+    private static PredictionEngine<CommitClassification, CommitClassificationPrediction> _predictionEngine;
+    private static ITransformer _trainedModel;
+    private static IDataView _trainingDataView;

```

Comment:
Als het goed is is `static` hier niet nodig, want we gebruik de service a.d.h.v. DI, als [scoped service](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-6.0#service-lifetimes), waardoor er op een gegeven moment (in een bepaalde scope basically) maar 1 instantie is van deze service.